### PR TITLE
Fix invert mask bugs in simulator.

### PR DESCRIPTION
**Pull request by dabacon**
Invert mask was not implemented for case where measurements were terminal and run was called.  Also fixes bug in invert mask is not completely specified.

### CODE REVIEW

**Comment by dabacon** [comment]
Fixes #1943

File: cirq/sim/simulator.py [hunk 1]
```diff
@@ -468,21 +468,31 @@ def sample_measurement_ops(
...
-                bounds.items()}
+
+        result = {
+            k: np.array([x[s:e] for x in indexed_sample
+                        ]) for k, (s, e) in bounds.items()
+        }
+        corrected_results = {}
+        for k, bits in result.items():
+            meas = meas_ops[k]
+            corrected_results[k] = (bits ^ meas.full_invert_mask())
```
[the remark this answers is not in the document]
Comment (dabacon) [reply cirq/sim/simulator.py]: sorry what?  Every loop is a different measurement with a different full_invert_mask

File: cirq/sim/simulator.py [hunk 2]
```diff
@@ -468,21 +468,31 @@ def sample_measurement_ops(
...
         indexed_sample = self.sample(all_qubits, repetitions)
-        return {k: np.array([x[s:e] for x in indexed_sample]) for k, (s, e) in
-                bounds.items()}
+
+        result = {
+            k: np.array([x[s:e] for x in indexed_sample
+                        ]) for k, (s, e) in bounds.items()
+        }
+        corrected_results = {}
+        for k, bits in result.items():
```
[the remark this answers is not in the document]
Comment (dabacon) [reply cirq/sim/simulator.py]: Done.

File: cirq/sim/simulator_test.py [hunk 3]
```diff
@@ -171,6 +171,21 @@ def test_step_sample_measurement_ops_repetitions():
...
+    measurement_ops = [
+        cirq.measure(q0, q1, invert_mask=(True,)),
+        cirq.measure(q2, invert_mask=(True,))
+    ]
+    step_result = FakeStepResult([q1])
+
+    measurements = step_result.sample_measurement_ops(measurement_ops)
+    np.testing.assert_equal(measurements, {
+        '0,1': [[True, True]],
+        '2': [[True]]
```
[the remark this answers is not in the document]
Comment (dabacon) [reply cirq/sim/simulator_test.py]: done

File: cirq/ops/common_gates.py [hunk 4]
```diff
@@ -479,6 +479,21 @@ def with_bits_flipped(self, *bit_positions: int) -> 'MeasurementGate':
...
+        padded by False.
+
+        Similarly if no invert_mask is supplies this returns the a tuple
+        of size equal to the number of qubits with all entries False.
+        """
+        mask = (self.invert_mask or self.num_qubits() * (False,))
+        deficit = self.num_qubits() - len(mask)
+        if deficit > 0:
+            mask += (False,) * deficit
+        return mask
```
[the remark this answers is not in the document]
Comment (cduck) [reply cirq/ops/common_gates.py]: This is fine for now.  All the simulators will need to be fixed to support qudits.  `invert_mask` may become "addition modulo number of levels".

File: cirq/google/engine/engine.py [hunk 5]
```diff
@@ -849,8 +849,8 @@ def __init__(self,
 
         Args:
             job_config: The JobConfig used to create the job.
-            job: A full Job Dict.
-            engine: An Engine instance associated with the same projct as the
+            job: A full Job Dict correspond to the QuantumJob proto.
+            engine: An Engine instance associated with the same project as the
```
Comment (cduck) [inline cirq/google/engine/engine.py]: This file looks like it was accidentally committed.

File: cirq/sim/simulator.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (cduck) [reply cirq/sim/simulator.py]: Nevermind.  I was thinking you were looping over repetitions.

File: cirq/sim/simulator.py [hunk 6]
```diff
@@ -468,21 +468,30 @@ def sample_measurement_ops(
...
-        return {k: np.array([x[s:e] for x in indexed_sample]) for k, (s, e) in
-                bounds.items()}
+
+        result = {
+            k: np.array([x[s:e] for x in indexed_sample
+                        ]) for k, (s, e) in bounds.items()
+        }
+        corrected_results = {
+            k: bits ^ meas_ops[k].full_invert_mask for k, bits in result.items()
+        }
```
Comment (dstrain115) [inline cirq/sim/simulator.py]: Can't we combine these two?  Something like this:

```
corrected_results = {
            k: np.array([x[s:e] for x in indexed_sample]) ^ meas_ops[k].full_invert_mask for k, (s, e) in bounds.items()
        }
```

Comment (dabacon) [reply cirq/sim/simulator.py]: that's way too complex for my liking.  generally if the comprehension has to span two lines I get nervous.

File: cirq/ops/common_gates.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (dabacon) [reply cirq/ops/common_gates.py]: that or any classical permutation.

File: cirq/sim/simulator.py (hunk 6 above)
Comment (cduck) [reply cirq/sim/simulator.py]: That's why I suggested using a helper function.

Comment (cduck) [reply cirq/sim/simulator.py]: It looks much better now.  It was confusing when it was half comprehension, half loop.

Comment (dstrain115) [reply cirq/sim/simulator.py]: Yeah, I am good with that. (I think I already approved, but LGTM)

Comment (dabacon) [reply cirq/sim/simulator.py]: @cduck LGTM?

**Pull request merged**